include/kernel.h

```diff
@@ -500,6 +500,8 @@ struct _thread_base {
 	/* this thread's entry in a timeout queue */
 	struct _timeout timeout;
 #endif
+
+	_wait_q_t join_waiters;
```

Comment:
going to look into making this an slist, I don't think we need a full priority queue since we do a wake-all instead of wake-one.